testsuite/pyproject.toml

```diff
@@ -0,0 +1,56 @@
+[build-system]
+# Minimum requirements for the build system to execute
+requires = [
+    "packaging",
+    "setuptools",
+    "wheel",
+]
+
+[project]
+name = "MDAnalysisTests"
+version = "2.1.0-dev0"
+license = {file = "LICENSE"}
+authors = [
+    {name = 'MDAnalysis Development Team', email = 'mdanalysis@numfocus.org'}
+]
+maintainers = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
+]
+requires-python = ">=3.7"
+dependencies = [
+    'MDAnalysis==2.1.0-dev0',
```

Comment:
I'm not super into the idea of having _yet-another_ point where we need to update the version numbers. Is there a way around this?

Reply:
Hmm, it is indeed pretty ugly from a design standpoint now to have separate main and test packages.

Is there as much of a rush to add a `pyproject.toml` for the test suite?

It would probably be nice to have the version dynamically set to match the same version of the main package. Not sure how to do that.

Reply:
A bit out of scope, but what other major Python projects package their test suite entirely separately from the main project? That might be a good reference point for "how should we handle this?"

If we were to start from scratch, I'd almost certainly vote against having separate packages, and instead use one of the popular "lazy loading/lazy download" packages to pull in test assets if there's an internet connection, otherwise skipping and doing a minimal suite only.

Reply:
For 3.0, getting rid of MDAnalysisTests would be a good aim. One thing to solve would be to keep the correct versions of test files in sync with the tests. Maybe download from the correct tag or commit in the repo? That might get a bit more tricky when you want to add/change test files in a branch... anyway, that would be something to think about and to learn from other packages that presumably already do this.

Which lazy-loading packages do you have in mind?

Do you have an example of a package that pulls in test files when needed?